L1Trigger/L1TTwinMux/src/L1TTwinMuxAlgorithm.cc

```diff
@@ -109,32 +116,42 @@ void L1TwinMuxAlgortithm::run(
       for (sector=0;sector<12; sector++ ){
         for (station=1; station<=4; station++){
 
-          dtts1=0; dtts2=0; rpcts1=0;
-
-          dtts1 = phiDigiscp.chPhiSegm1(wheel,station,sector,bx);
-          dtts2 = phiDigiscp.chPhiSegm2(wheel,station,sector,bx);
-          rpcts1 = rpcPhiDigis.chPhiSegm1(wheel,station,sector,bx);
-
-            if(!onlyRPC) {
-              if(!dtts1 && !dtts2 && !rpcts1 ) continue;
+         // dtts1=0; dtts2=0; rpcts1=0;
+
+         // dtts1 = phiDigiscp.chPhiSegm1(wheel,station,sector,bx);
+         // dtts2 = phiDigiscp.chPhiSegm2(wheel,station,sector,bx);
+         // rpcts1 = rpcPhiDigis.chPhiSegm1(wheel,station,sector,bx);
+           dtts1.reset(new L1MuDTChambPhDigi(0,0,0,0,0,0,7,0,0,0));
+           dtts2.reset(new L1MuDTChambPhDigi(0,0,0,0,0,0,7,0,0,0));
+           rpcts1.reset(new L1MuDTChambPhDigi(0,0,0,0,0,0,7,0,0,0));
+           if (phiDigiscp.chPhiSegm1(wheel,station,sector,bx)) 
+              dtts1.reset(new L1MuDTChambPhDigi(phiDigiscp.chPhiSegm1(wheel,station,sector,bx)->bxNum(),phiDigiscp.chPhiSegm1(wheel,station,sector,bx)->whNum(),phiDigiscp.chPhiSegm1(wheel,station,sector,bx)->scNum(),phiDigiscp.chPhiSegm1(wheel,station,sector,bx)->stNum(),phiDigiscp.chPhiSegm1(wheel,station,sector,bx)->phi(),phiDigiscp.chPhiSegm1(wheel,station,sector,bx)->phiB(),phiDigiscp.chPhiSegm1(wheel,station,sector,bx)->code(),phiDigiscp.chPhiSegm1(wheel,station,sector,bx)->Ts2Tag(),phiDigiscp.chPhiSegm1(wheel,station,sector,bx)->BxCnt(),phiDigiscp.chPhiSegm1(wheel,station,sector,bx)->RpcBit()));
+           if (phiDigiscp.chPhiSegm2(wheel,station,sector,bx))
+              dtts2.reset(new L1MuDTChambPhDigi(phiDigiscp.chPhiSegm2(wheel,station,sector,bx)->bxNum(),phiDigiscp.chPhiSegm2(wheel,station,sector,bx)->whNum(),phiDigiscp.chPhiSegm2(wheel,station,sector,bx)->scNum(),phiDigiscp.chPhiSegm2(wheel,station,sector,bx)->stNum(),phiDigiscp.chPhiSegm2(wheel,station,sector,bx)->phi(),phiDigiscp.chPhiSegm2(wheel,station,sector,bx)->phiB(),phiDigiscp.chPhiSegm2(wheel,station,sector,bx)->code(),phiDigiscp.chPhiSegm2(wheel,station,sector,bx)->Ts2Tag(),phiDigiscp.chPhiSegm2(wheel,station,sector,bx)->BxCnt(),phiDigiscp.chPhiSegm2(wheel,station,sector,bx)->RpcBit()));
+           if (rpcPhiDigis.chPhiSegm1(wheel,station,sector,bx))
+              rpcts1.reset(new L1MuDTChambPhDigi(rpcPhiDigis.chPhiSegm1(wheel,station,sector,bx)->bxNum(),rpcPhiDigis.chPhiSegm1(wheel,station,sector,bx)->whNum(),rpcPhiDigis.chPhiSegm1(wheel,station,sector,bx)->scNum(),rpcPhiDigis.chPhiSegm1(wheel,station,sector,bx)->stNum(),rpcPhiDigis.chPhiSegm1(wheel,station,sector,bx)->phi(),rpcPhiDigis.chPhiSegm1(wheel,station,sector,bx)->phiB(),rpcPhiDigis.chPhiSegm1(wheel,station,sector,bx)->code(),rpcPhiDigis.chPhiSegm1(wheel,station,sector,bx)->Ts2Tag(),rpcPhiDigis.chPhiSegm1(wheel,station,sector,bx)->BxCnt(),rpcPhiDigis.chPhiSegm1(wheel,station,sector,bx)->RpcBit()));
+
+          if(!onlyRPC) {
```

Comment:
Can this be made the `else` of the `if` in line 151?

Reply:
it is probably better if this is raw pointer as well with the same arguments